### PR TITLE
Auto Delete after 15 days service added

### LINKED ISSUE #798: Auto Delete Expired Downloads
Write a Worker Request which will delete the expired downloads usually they have a life of 15 days
So after 15 days, a video should be deleted automatically.

### PR DESCRIPTION
**Pull request by prathamk22**
Fixes #798 
 It will delete downloads automatically after 15 days

Screenshots for the change:

### CODE REVIEW

**Comment by aggarwalpulkit596** [comment]
@prathamk22 any updates on this ?

**Comment by prathamk22** [comment]
Sir actually universities are taking internal exams from past 2 days. Will push in asap.

**Comment by aggarwalpulkit596** [comment]
Are you updating the db as well after deletion of files ?

**Comment by prathamk22** [comment]
Yes sir it is updating db too

File: app/src/main/java/com/codingblocks/cbonlineapp/util/DeleteDownloadedFiles.kt [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            if (item.date != today)
+                return@forEach
+            val file =
+                applicationContext.getExternalFilesDir(Environment.getDataDirectory().absolutePath)
+            val folderFile = File(
+                file,
+                "/${item.videoId}"
+            )
+            MediaUtils.deleteRecursive(folderFile)
+            contentDao.updateContentWithVideoId(item.videoId, 0)
```
Comment (prathamk22) [inline app/src/main/java/com/codingblocks/cbonlineapp/util/DeleteDownloadedFiles.kt]: Here sir, it is updating the db when video is deleted

**Comment by aggarwalpulkit596** [comment]
There is no need to add another modal for download data you can get the time when it was downloaded or last modified and do the work depending upon that time

**Comment by prathamk22** [comment]
@aggarwalpulkit596 changes done as per feedback sir.

**Comment by aggarwalpulkit596** [comment]
Will take a look

**Review by aggarwalpulkit596 — APPROVED** [review]

File: app/src/main/java/com/codingblocks/cbonlineapp/util/DeleteDownloadedFiles.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    val contentDao: ContentDao by inject()
+
+    override suspend fun doWork(): Result {
+        val list:List<ContentModel> = contentDao.getDownloads(true)
+
+        list.forEach {item->
+            val folderFile = File(applicationContext.getExternalFilesDir(Environment.getDataDirectory().absolutePath), "/${item.contentLecture.lectureId}")
+            val fileDate = Date(folderFile.lastModified())
+            val expiringDate = Date(fileDate.time + 15)
```
Comment (aggarwalpulkit596) [inline app/src/main/java/com/codingblocks/cbonlineapp/util/DeleteDownloadedFiles.kt:25]: @prathamk22 
```
            val calendar = Calendar.getInstance()
            calendar.time = Date(fileDate.time)
            calendar.add(Calendar.DATE, 14)
            val expiringDate = calendar.time
```

it was supposed to be done like this

Comment (prathamk22) [reply app/src/main/java/com/codingblocks/cbonlineapp/util/DeleteDownloadedFiles.kt:25]: Okay sir.

**Pull request merged**